### DISCUSSION TITLE
Priority in MQ?

### DISCUSSION THREAD

**Discussion by hosseiin1993** [python-telegram-bot/python-telegram-bot#2265] [Feature Suggestions]
Hi 
have you ever thought about setting priority for bot messages in MQ?
in my case, I'm sending a batch of news to 100k users in an exact time, and my bot response time increases
so I want to respond to user's interactions with higher priority than sending my news list.

**Comment by Bibo-Joshi** [comment]
Hi. That's an interesting suggestion!

I guess that should be rather easy to implement using a `PriorityQueue`. In fact, refactoring MQ is a current project (see #2160) and we can think about including this feature. We would just need to add a parameter `priority` to `DelayQueue` that's used if the `DelayQueue` has a parent. Then you could do

```python
MQ.add_delay_queue('low_priority_queue', DelayQueue(…, parent='default_queue', priorty=100))
```
and put your batch notifications through that queue.

**Reply by hosseiin1993** [reply]
that would be awesome!
thanks

**Reply by Bibo-Joshi** [reply]
Just an FYI: Refactoring MQ was [postponed](https://github.com/python-telegram-bot/python-telegram-bot/issues/2139#issuecomment-753496322) , but we will defenitely keep the idea in mind :)
